game-core/src/main/java/games/strategy/engine/framework/ui/SaveGameFileChooser.java

```diff
@@ -46,28 +45,34 @@
      *             once no stable client will ever request this auto-save).
      */
     @Deprecated
-    AUTOSAVE2(getHeadlessAutoSaveFileName()),
+    AUTOSAVE2(getHeadlessAutoSaveFile()),
 
-    AUTOSAVE_ODD(getOddRoundAutoSaveFileName(true)),
+    AUTOSAVE_ODD(getOddRoundAutoSaveFile(true)),
 
-    AUTOSAVE_EVEN(getEvenRoundAutoSaveFileName(true));
+    AUTOSAVE_EVEN(getEvenRoundAutoSaveFile(true));
 
     private final String fileName;
 
-    AUTOSAVE_TYPE(final String fileName) {
-      this.fileName = fileName;
+    AUTOSAVE_TYPE(final File file) {
+      this.fileName = file.getName();
     }
 
-    public String getFileName() {
-      return fileName;
+    public File getFile() {
+      return getAutoSaveFile(fileName);
```

Comment:
Is there a specific reason only a string is stored instead of a file object? 
I could imagine you did this either in order to ensure immutability or to make this enum dynamic in the sense that the savegame directory may change?

Reply:
Yes, sorry, I should have documented this in the PR.

`AUTOSAVE_TYPE` is serialized via RMI calls.  I _assumed_ that the `serialVersionUID` field for enums is auto-generated by the compiler, and thus changing the fields would cause a different `serialVersionUID` to be generated, which would break compatibility with an older peer on the other end.

However, it appears that the compiler always generates a `serialVersionUID` value of zero regardless of the field count and types.  And since enums are not truly deserialized (there's an effective `readResolve()` that guarantees to always return the single in-process instance of each enum constant, thus ignoring any fields that may have been serialized by the sender), there should be no compatibility issue.  I verified this by having a client from this branch attempt to load auto-saves against a 10663 bot, and everything appeared to work correctly.  87f646b1f